plugins/SitesManager/API.php

```diff
@@ -391,6 +391,14 @@ public function getSitesWithAdminAccess($fetchAliasUrls = false, $pattern = fals
             }
         }
 
+        if ($updateDeleteSiteExplanation) {
+            foreach ($sites as &$site) {
+                $deleteSiteExplanation = Piwik::translate('SitesManager_DeleteSiteExplanation');
+                Piwik::postEvent('SitesManager.updateDeleteSiteExplanation', [$site['idsite'], &$deleteSiteExplanation]);
+                $site['delete_site_explanation'] = $deleteSiteExplanation;
+            }
+        }
+
```

Comment:
It's imho a bad choice to update/change the API only to inject something that will only be visible in the UI.
For that it's better to introduce a new UI / JavaScript event another plugin can listen to.
Is there are reason why you didn't use `Matomo.postEvent()` directly in the `SiteFields` component?

Reply:
@sgiehl Changed to fetch the content via an API call only when delete action is triggered